### PR TITLE
Remove privileged for `inspektor-gadget` container.

### PR DESCRIPTION
**Pull request by eiffel-fl**
Hi.

This PR addresses #128.
Basically, we need to add several capabilities (in addition to the [default ones](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_atomic_host/7/html/container_security_guide/linux_capabilities_and_seccomp)):
1. `CAP_SYS_RESOURCE`: `gadgetracermanager` calls `setrlimit` with an infinite value.
2. `CAP_SYS_ADMIN`: To interact with BPF.
3. `CAP_SYSLOG`: To get kernel symbols addresses in `/proc/kallsyms`.
4. `CAP_NET_ADMIN`: To be able to create BPF link with `BPF_PROG_TYPE_CGROUP_SKB` program type.

I tested all the python gadgets (expect `bindsnoop` which was CORE tested).
I used three kernels: 5.10.70 (with `kind`), 5.10.37 (with `kvm2`) and 5.4.40 (with `kvm2`).

<details>
  <summary>Here are the output for all of them:</summary>

* `bindsnoop`:
```bash
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget bindsnoop                                     francis/remove-privileged *% u=
NODE             NAMESPACE        POD              CONTAINER        PID     COMM             RET PROTO OPTS  IF   PORT  ADDR
# Jump to an other terminal.
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl run --restart=Never --image=debian mypod -- sleep inf  francis/remove-privileged *% u=
pod/mypod created
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl exec -ti mypod -- bash                                 francis/remove-privileged *% u=
root@mypod:/# apt update && apt install netcat
...
root@mypod:/# netcat -l 80
# Go back to first terminal
Tracing binds ... Hit Ctrl-C to end
NODE             NAMESPACE        POD              CONTAINER             PID COMM         PROT ADDR            PORT   OPTS IF
minikube         default          mypod            mypod               31811 netcat       UNKN 0.0.0.0            80 ...Rr  0
```
* `biotop`:
```bash
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget biotop --all-namespaces --node ip-10-0-30-247 francis/remove-privileged *% u=

FATA[0000] biotop only works with --all-namespaces       args="[]" command="kubectl-gadget biotop"
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl exec -ti -n kube-system gadget-8bdrj -- bash
root@minikube:/# /usr/bin/gadgets/biotop
15:29:32 loadavg: 1.06 1.55 1.11 2/415 17657

PID    COMM             D MAJ MIN DISK       I/O  Kbytes  AVGms
1827   etcd             W 253 0   vda          9      36   0.30
0                       R 253 0   vda          6       0  44.91

```
* `capabilities`:
```bash
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl apply -f docs/examples/app-set-priority.yaml           francis/remove-privileged *% u=
deployment.apps/set-priority created
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl logs -lname=set-priority                               francis/remove-privileged *% u=
nice: setpriority(-20): Permission denied
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget capabilities                                  francis/remove-privileged *% u=
NODE             NAMESPACE        POD              CONTAINER        TIME      UID    PID    COMM             CAP  NAME                 AUDIT 
minikube         default          set-priority-5646554d9d-zpfqt set-priority     16:08:53  0      42642  nice             23   CAP_SYS_NICE         1     
minikube         default          set-priority-5646554d9d-zpfqt set-priority     16:08:58  0      42699  nice             23   CAP_SYS_NICE         1 
```
* `dns`:
```bash
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl create ns demo                                         francis/remove-privileged *% u=
namespace/demo created
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget dns -n demo                                   francis/remove-privileged *% u=
POD                            TYPE      NAME
# Jump to another terminal.
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl -n demo run mypod -it --image=praqma/network-multitool -- /bin/sh
If you don't see a command prompt, try pressing enter.
/ # nslookup www.wikipedia.org
Server:         10.96.0.10
Address:        10.96.0.10#53
...
# Comeback to first terminal.
POD                            TYPE      NAME
mypod                          OUTGOING  www.wikipedia.org.demo.svc.cluster.local.
mypod                          OUTGOING  www.wikipedia.org.svc.cluster.local.
mypod                          OUTGOING  www.wikipedia.org.cluster.local.
mypod                          OUTGOING  www.wikipedia.org.
mypod                          OUTGOING  dyna.wikimedia.org.
```
* `execsnoop`:
```bash
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl apply -f docs/examples/ds-myapp.yaml                   francis/remove-privileged *% u=
daemonset.apps/myapp1-pod created
daemonset.apps/myapp2-pod created
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl get pod --show-labels -o wide                          francis/remove-privileged *% u=
NAME               READY   STATUS    RESTARTS   AGE   IP           NODE       NOMINATED NODE   READINESS GATES   LABELS
myapp1-pod-2scqp   1/1     Running   0          18s   172.17.0.3   minikube   <none>           <none>            controller-revision-hash=d4b99fcc,myapp=app-one,name=myapp1-pod,pod-template-generation=1,role=demo
myapp2-pod-wp7cg   1/1     Running   0          18s   172.17.0.4   minikube   <none>           <none>            controller-revision-hash=7cd97b95c,myapp=app-two,name=myapp2-pod,pod-template-generation=1,role=demo
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget execsnoop --selector role=demo --node minikube
NODE             NAMESPACE        POD              CONTAINER        PCOMM            PID    PPID   RET ARGS
minikube         default          myapp1-pod-xw5g4 myapp1-pod       true             66308  66131    0 /bin/true
minikube         default          myapp1-pod-xw5g4 myapp1-pod       date             66314  66131    0 /bin/date
minikube         default          myapp1-pod-xw5g4 myapp1-pod       cat              66317  66131    0 /bin/cat /proc/version
minikube         default          myapp1-pod-xw5g4 myapp1-pod       sleep            66318  66131    0 /bin/sleep 1
minikube         default          myapp1-pod-xw5g4 myapp1-pod       true             66350  66131    0 /bin/true
minikube         default          myapp1-pod-xw5g4 myapp1-pod       date             66351  66131    0 /bin/date
minikube         default          myapp1-pod-xw5g4 myapp1-pod       cat              66352  66131    0 /bin/cat /proc/version
minikube         default          myapp1-pod-xw5g4 myapp1-pod       sleep            66353  66131    0 /bin/sleep 1
^C
Terminating...
```
* `moutsnoop`:
```bash
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget mountsnoop                                    francis/remove-privileged *% u=
NODE             NAMESPACE        POD              CONTAINER        COMM             PID     TID     MNT_NS      CALL
# Jump to other terminal
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl run mnt-app --image=busybox -- sleep inf               francis/remove-privileged *% u=
pod/mnt-app created
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl exec -ti mnt-app -- mount /mnt /mnt                    francis/remove-privileged *% u=
mount: mounting /mnt on /mnt failed: No such file or directory
command terminated with exit code 255
# Go back to first terminal.
minikube         default          mnt-app          mnt-app          mount            71594   71594   4026532788  mount("/mnt", "/mnt", "ext3", MS_SILENT, "") = -ENOENT
minikube         default          mnt-app          mnt-app          mount            71594   71594   4026532788  mount("/mnt", "/mnt", "ext2", MS_SILENT, "") = -ENOENT
minikube         default          mnt-app          mnt-app          mount            71594   71594   4026532788  mount("/mnt", "/mnt", "ext4", MS_SILENT, "") = -ENOENT
minikube         default          mnt-app          mnt-app          mount            71594   71594   4026532788  mount("/mnt", "/mnt", "vfat", MS_SILENT, "") = -ENOENT
minikube         default          mnt-app          mnt-app          mount            71594   71594   4026532788  mount("/mnt", "/mnt", "msdos", MS_SILENT, "") = -ENOENT
minikube         default          mnt-app          mnt-app          mount            71594   71594   4026532788  mount("/mnt", "/mnt", "iso9660", MS_SILENT, "") = -ENOENT
minikube         default          mnt-app          mnt-app          mount            71594   71594   4026532788  mount("/mnt", "/mnt", "fuseblk", MS_SILENT, "") = -ENOENT
minikube         default          mnt-app          mnt-app          mount            71594   71594   4026532788  mount("/mnt", "/mnt", "xfs", MS_SILENT, "") = -ENOENT
```
* `network-policy`:
```bash
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl create ns demo                                         francis/remove-privileged *% u=
namespace/demo created
# In an other terminal:
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget network-policy monitor --namespaces demo --output ./networktrace.log
# Jump back to first one:
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl apply -f docs/examples/disable-psp-demo.yaml           francis/remove-privileged *% u=
clusterrole.rbac.authorization.k8s.io/disable-psp-demo created
clusterrolebinding.rbac.authorization.k8s.io/disable-psp-demo created
francis@machine:~/Codes/kinvolk/inspektor-gadget$ wget -O network-policy-demo.yaml https://raw.githubusercontent.com
...
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl apply -f network-policy-demo.yaml -n demo
...
# Wait until almost all are running.
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl get pod -n demo                                        francis/remove-privileged *% u=
NAME                                     READY   STATUS             RESTARTS      AGE
adservice-85c4b6d76f-bvxn9               0/1     Pending            0             3m19s
cartservice-5df868bd88-bqxdh             0/1     Completed          2 (30s ago)   3m39s
checkoutservice-5fc5745cf9-jkcm6         1/1     Running            0             3m55s
currencyservice-f97fd6665-xdr77          0/1     Running            0             3m32s
emailservice-df676867c-twrh5             0/1     CrashLoopBackOff   4 (20s ago)   3m55s
frontend-95cfb9957-tpb66                 1/1     Running            0             3m52s
loadgenerator-7b99f6ff4c-qcsrn           0/1     CrashLoopBackOff   1 (13s ago)   3m35s
paymentservice-646496b76f-rtqnc          1/1     Running            0             3m46s
productcatalogservice-fffbfb9f-55vwk     1/1     Running            0             3m43s
recommendationservice-79f656db6c-gnqfq   0/1     CrashLoopBackOff   4 (10s ago)   3m54s
redis-cart-5b569cd47-xrrlh               0/1     Pending            0             3m21s
shippingservice-75b96c4855-cswjm         0/1     Pending            0             3m28s
# Go back to second terminal
^C
Stopping...
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget network-policy report --input ./networktrace.log > network-policy.yaml
francis@machine:~/Codes/kinvolk/inspektor-gadget$ head network-policy.yaml                                       francis/remove-privileged *% u=
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  creationTimestamp: null
  name: checkoutservice-network
  namespace: demo
spec:
  ingress:
  - from:
    - ipBlock:
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl apply -f network-policy.yaml                           francis/remove-privileged *% u=
networkpolicy.networking.k8s.io/checkoutservice-network created
networkpolicy.networking.k8s.io/frontend-network created
networkpolicy.networking.k8s.io/loadgenerator-network created
networkpolicy.networking.k8s.io/productcatalogservice-network created
```
* `opensnoop`:
```bash
kubectl run --restart=Never -ti --image=busybox mypod -- sh -c 'while /bin/true ; do whoami ; sleep 3 ; done'

If you don't see a command prompt, try pressing enter.
root
root
root
# Jump to an other terminal.
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget opensnoop --podname mypod                     francis/remove-privileged *% u=
NODE             NAMESPACE        POD              CONTAINER        PID    COMM               FD ERR PATH
minikube         default          mypod            mypod            74942  whoami              3   0 /etc/passwd
minikube         default          mypod            mypod            74995  whoami              3   0 /etc/passwd
minikube         default          mypod            mypod            75010  whoami              3   0 /etc/passwd
```
* `process-collector`:
```bash
WARN[0002] Error getting traces from node "kind-control-plane": failed to attach BPF iterator: detect support for bpf_link: operation not permitted  args="[]" command="kubectl-gadget process-collector"
FATA[0002] Error getting traces from all nodes           args="[]" command="kubectl-gadget process-collector"
```
* `process-collector`:
```bash
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl create ns test-socketcollector                         francis/remove-privileged *% u=
namespace/test-socketcollector created
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl run --restart=Never -n test-socketcollector --image=nginx nginx-app --port=80
pod/nginx-app created
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget process-collector start -n test-socketcollector
NAMESPACE               POD          CONTAINER    COMM     PID     
test-socketcollector    nginx-app    nginx-app    nginx    4941    
test-socketcollector    nginx-app    nginx-app    nginx    4980    
test-socketcollector    nginx-app    nginx-app    nginx    4981
```
* `profile`:
```bash
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl run --restart=Never --image=busybox random -- sh -c 'cat /dev/urandom > /dev/null'
# Jump to other terminal:
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget profile --podname random -K
# Wait a bit then Ctrl + C:
^C
Terminating...
Terminating...
cat;entry_SYSCALL_64_after_hwframe;do_syscall_64;syscall_slow_exit_work;__audit_syscall_exit;kfree;kfree 1
cat;entry_SYSCALL_64_after_hwframe;do_syscall_64;syscall_trace_enter;ktime_get_coarse_real_ts64;ktime_get_coarse_real_ts64 1
cat;entry_SYSCALL_64_after_hwframe;do_syscall_64;__x64_sys_write;__x64_sys_write 1
cat;entry_SYSCALL_64_after_hwframe;do_syscall_64;ksys_read;vfs_read;urandom_read;extract_crng;extract_crng 1
cat;entry_SYSCALL_64_after_hwframe;do_syscall_64;ksys_write;vfs_write;fsnotify;fsnotify 1
cat;entry_SYSCALL_64_after_hwframe;do_syscall_64;ksys_read;vfs_read;urandom_read;urandom_read 1
cat;entry_SYSCALL_64_after_hwframe;do_syscall_64;ksys_read;vfs_read;security_file_permission;selinux_file_permission;selinux_file_permission 1
cat;entry_SYSCALL_64_after_hwframe;do_syscall_64;syscall_trace_enter;syscall_trace_enter 1
cat;entry_SYSCALL_64_after_hwframe;do_syscall_64;ksys_read;vfs_read;urandom_read;_extract_crng;_extract_crng 1
cat;entry_SYSCALL_64_after_hwframe;do_syscall_64;ksys_read;vfs_read;urandom_read;urandom_read 2
cat;entry_SYSCALL_64_after_hwframe;do_syscall_64;do_syscall_64 2
cat;entry_SYSCALL_64_after_hwframe;do_syscall_64;ksys_read;vfs_read;urandom_read;_copy_to_user;copy_user_generic_string;copy_user_generic_string 2
cat 5
cat;entry_SYSCALL_64_after_hwframe;do_syscall_64;ksys_read;vfs_read;urandom_read;_raw_spin_unlock_irqrestore;_raw_spin_unlock_irqrestore 45
```
* `seccomp`:
```bash
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl apply -f docs/examples/seccomp/basic.yaml              francis/remove-privileged *% u=
namespace/seccomp-demo created
configmap/app-script created
service/hello-python-service created
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl apply -f docs/examples/seccomp/unconfined.yaml         francis/remove-privileged *% u=
pod/hello-python created
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl apply -f docs/examples/seccomp/trace-status.yaml       francis/remove-privileged *% u=
trace.gadget.kinvolk.io/seccomp created
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl annotate -n gadget trace/seccomp gadget.kinvolk.io/operation=start
trace.gadget.kinvolk.io/seccomp annotated
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl port-forward service/hello-python-service -n seccomp-demo 8081:6000 &
[1] 144471
francis@machine:~/Codes/kinvolk/inspektor-gadget$ Forwarding from 127.0.0.1:8081 -> 80                           francis/remove-privileged *% u=
Forwarding from [::1]:8081 -> 80
francis@machine:~/Codes/kinvolk/inspektor-gadget$  curl localhost:8081
Handling connection for 8081
Hello World!
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl annotate --overwrite -n gadget trace/seccomp gadget.kinvolk.io/operation=generate
trace.gadget.kinvolk.io/seccomp annotated
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl get -n gadget trace/seccomp -o custom-columns=Status:.status
Status
map[output:{
  "defaultAction": "SCMP_ACT_ERRNO",
  "architectures": [
    "SCMP_ARCH_X86_64",
    "SCMP_ARCH_X86",
    "SCMP_ARCH_X32"
  ],
  "syscalls": [
    {
      "names": [
        "accept4",
        "close",
        "connect",
        "epoll_ctl",
        "epoll_wait",
        "fstat",
        "getsockname",
        "getsockopt",
        "gettimeofday",
        "ioctl",
        "poll",
        "read",
        "recvfrom",
        "setsockopt",
        "socket",
        "stat",
        "wait4",
        "write",
        "writev"
      ],
      "action": "SCMP_ACT_ALLOW"
    }
  ]
} state:Started]
```
* `socket-collector`:
```
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl create ns test-socketcollector                         francis/remove-privileged *% u=
namespace/test-socketcollector created
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl run --restart=Never -n test-socketcollector --image=nginx nginx-app --port=80
pod/nginx-app created
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget socket-collector start -n test-socketcollector
NODE        NAMESPACE               POD          PROTOCOL    LOCAL         REMOTE       STATUS
minikube    test-socketcollector    nginx-app    TCP         0.0.0.0:80    0.0.0.0:0    LISTEN
```
* `tcpconnect`:
```bash
# In a first terminal:
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget tcpconnect --podname mypod                    francis/remove-privileged *% u=
Tracing connect ... Hit Ctrl-C to end
NODE             NAMESPACE        POD              CONTAINER        PID    COMM         IP SADDR            DADDR            DPORT
# Open another:
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl run --restart=Never -ti --image=busybox mypod -- sh -c 'while true; do wget -q -O /dev/null -T 3 http://1.1.1.1; sleep 5; done'
If you don't see a command prompt, try pressing enter.
wget: note: TLS certificate validation not implemented
wget: note: TLS certificate validation not implemented
wget: note: TLS certificate validation not implemented
# Go back to first one.
NODE             NAMESPACE        POD              CONTAINER        PID    COMM         IP SADDR            DADDR            DPORT 
minikube         default          mypod            mypod            80344  wget         4  172.17.0.5       1.1.1.1          80     
minikube         default          mypod            mypod            80344  wget         4  172.17.0.5       1.1.1.1          443    
minikube         default          mypod            mypod            80438  wget         4  172.17.0.5       1.1.1.1          80     
minikube         default          mypod            mypod            80438  wget         4  172.17.0.5       1.1.1.1          443    
minikube         default          mypod            mypod            80503  wget         4  172.17.0.5       1.1.1.1          80     
minikube         default          mypod            mypod            80503  wget         4  172.17.0.5       1.1.1.1          443
```
* `tcptop`:
```
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl run --restart=Never --image=busybox mypod -- sh -c 'while /bin/true ; do wget -O - https://kinvolk.io ; sleep 3 ; done'
pod/mypod created
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl get pod -o wide                                        francis/remove-privileged *% u=
NAME     READY   STATUS    RESTARTS   AGE   IP           NODE       NOMINATED NODE   READINESS GATES
mypod    1/1     Running   0          9s    172.17.0.5   minikube   <none>           <none>
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl gadget tcptop --node minikube --podname mypod
18:09:43 loadavg: 1.30 1.56 1.75 2/484 148719

PID    COMM         LADDR                 RADDR                  RX_KB  TX_KB
148663 148663       172.17.0.5:54368      172.67.196.142:443        35      0
```
* `traceloop`:
```bash
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl run --restart=Never -ti --image=busybox mypod -- sh -c 'RANDOM=output ; echo "3*7*2" | bc > /tmp/file-$RANDOM ; cat /tmp/file-$RANDOM'
cat: can't open '/tmp/file-3240': No such file or directory
pod default/mypod terminated (Error)
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget traceloop list                                francis/remove-privileged *% u=
POD      PODUID      INDEX    TRACEID             CONTAINERID    STATUS    
mypod    87fc705c    0        000000f59abaf216    b87235d3       terminated less than a second ago
```  
</details>

To summarize:
| gadget | status | Note |
| --------- | -------- | --------- |
| bindsnoop | OK |  |
| biotop | OK |  |
| capabilities | OK |  |
| dns | OK |  |
| execsnoop | OK |  |
| mountsnoop | OK |  |
| network-policy | OK |  |
| opensnoop | OK |  |
| process-collector | OK | |
| profile | OK |  |
| seccomp | OK |  |
| socket-collector | OK | |
| tcpconnect | OK | |
| tcptop | OK |  |
| traceloop | OK | OK locally, not in CI |

Best regards.

### CODE REVIEW

**Comment by eiffel-fl** [comment]
Regarding the CI failing, here are the results for different configurations:
* `minikube kvm kernel 5.10`: OK
```
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl describe pod -n kube-system gadget-66qhw | grep hook   francis/remove-privileged *% u=
Annotations:  inspektor-gadget.kinvolk.io/option-hook-mode: fanotify
francis@machine:~/Codes/kinvolk/inspektor-gadget$ kubectl run --restart=Never -ti --image=busybox mypod -- sh -c 'RANDOM=output ; echo "3*7*2" | bc > /tmp/file-$RANDOM ; cat /tmp/file-$RANDOM'
cat: can't open '/tmp/file-3240': No such file or directory
pod default/mypod terminated (Error)
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget traceloop list                                francis/remove-privileged *% u=
POD      PODUID      INDEX    TRACEID             CONTAINERID    STATUS    
mypod    40bc1946    0        00000049e69e2610    c7cd1513       terminated 8 seconds ago
francis@machine:~/Codes/kinvolk/inspektor-gadget$ ./kubectl-gadget deploy | grep capa                            francis/remove-privileged *% u=
          capabilities:
              # if (!capable(CAP_SYS_ADMIN))
              # We need this capability to get addresses from /proc/kallsyms.
```
* `minikube kvm kernel 5.4`: KO (because this kernel does not permit `fanotify`)
```
Gadget Tracer Manager hook mode: fanotify
BTF is not available: Trying BTFHub
Trying to download vmlinux from https://github.com/aquasecurity/btfhub/raw/main/buildroot/2020.02.2/x86_64/5.4.40.btf.tar.xz
vmlinux not found
Starting the Gadget Tracer Manager in the background...
traceloop version 6f4efc6
{"level":"info","msg":"Starting pod informer","time":"2021-10-21T12:59:17Z"}
I1021 12:59:17.681955    5138 podinformer.go:282] Starting Pod controller
time="2021-10-21T12:59:17Z" level=info msg="Creating BPF map: /sys/fs/bpf/gadget/containers"
time="2021-10-21T12:59:17Z" level=info msg="GadgetTracerManager: hook mode: fanotify"
time="2021-10-21T12:59:17Z" level=fatal msg="failed to create server cannot start runc fanotify: fanotify: init error, function not implemented"
{"level":"info","msg":"BPF Tracer ready","time":"2021-10-21T12:59:23Z"}
{"level":"info","msg":"Starting proc informer","time":"2021-10-21T12:59:23Z"}
{"level":"info","msg":"Starting annotation publisher","time":"2021-10-21T12:59:23Z"}
{"level":"info","msg":"Starting updater loop","time":"2021-10-21T12:59:23Z"}
{"containerID":"","idx":1,"level":"info","msg":"Deleting tracelet","namespace":"","podname":"","poduid":"","time":"2021-10-21T12:59:40Z"}
```
* `kind docker kernel 5.10`: KO
```
time="2021-10-21T13:07:59Z" level=info msg="Serving on gRPC socket /run/gadgettracermanager.socket"
time="2021-10-21T13:08:00Z" level=info msg="Starting trace controller manager"
invalid guessing state while guessing uts namespace, got checking expected checked
```
